### ISSUE TITLE
childCombinator does breaks when there is an unclosed tag

### ISSUE THREAD

**Issue by gajus** [cheeriojs/cheerio#971]
Example:

```js
const input = `<!DOCTYPE html>
<head>
</head>
<body>
  <div id="headerInner">
    <h3>Welcome to WTW Cinemas</h2>

    <ul>
      <li>a</li>
      <li>b</li>
      <li>c</li>
    </ul>
  </div>
</body>
</html>`;

const cheerio = require('cheerio');

console.log( cheerio.load(input)('#headerInner > ul > li').length );

```

This produces output `0`. Expected output is `3`.

Example:

```js
const input = `<!DOCTYPE html>
<head>
</head>
<body>
  <div id="headerInner">
    <ul>
      <li>a</li>
      <li>b</li>
      <li>c</li>
    </ul>
  </div>
</body>
</html>`;

const cheerio = require('cheerio');

console.log( cheerio.load(input)('#headerInner > ul > li').length );

```

This produces output `3`. Expected output is `3`.

**Comment by gajus** [comment]
Ah, I just noticed that the opening and closing tags do not match:

```html
<h3>Welcome to WTW Cinemas</h2>

```

Thats obviously an error in source document. However, the above selector does not work as expected in the browser. Should it not work in Cheerio?

**Comment by fb55** [comment]
If it does not work in the browser, why should it work in cheerio?

**Comment by gajus** [comment]
@fb55 should this be marked as a bug?

**Comment by fb55** [comment]
Fixed by #985
